elide-async/src/main/java/com/yahoo/elide/async/service/AsyncQueryThread.java

```
@@ -147,14 +147,21 @@ else if (queryObj.getQueryType().equals(QueryType.GRAPHQL_V1_0)) {
      */
     protected Integer calculateRecordsJSON(String jsonStr) {
         Integer rec = null;
-        JsonElement jsonTree = JsonParser.parseString(jsonStr);
-        if (!jsonTree.isJsonObject()) {
-            return null;
-        }
-        JsonObject jsonObject = jsonTree.getAsJsonObject();
-        JsonElement jsonElement = jsonObject.get("data");
-        if (jsonElement.isJsonArray()) {
-            rec = jsonElement.getAsJsonArray().size();
+        try {
+            ObjectMapper objectMapper = new ObjectMapper();
+            JsonNode jsonNode = objectMapper.readTree(jsonStr);
+            if (!jsonNode.isObject()) {
+                return null;
+            }
+             jsonNode = jsonNode.get("data");
+            if (jsonNode.isArray()) {
+                rec = jsonNode.size();
+            }
+
+        } catch (JsonMappingException e) {
```

no need to catch exceptions. just let them propagate.